### ISSUE TITLE
HiCanu on metagenome dataset produces poor assembly 

### ISSUE THREAD

**Issue by Somebodyatthdoor** [marbl/canu#2126]
Hi,
I’ve been trying to run HiCanu on some Hifi metagenomic data from chicken gut samples. Despite following the recommended options for metagenomes, I have been struggling to generate a good quality assembly. I am using Canu v2.2 on linux. 

In comparison to hifiasm-meta and metaflye, Hicanu is producing a far smaller assembly. HiCanu creates an assembly of 27.5mb, metaflye = 473mb, hifiasm-meta = 704mb. My original gzipped fastq file is 13gb. 

When I then try and make genome bins from these assemblies using metabat2 I get a far higher number of bins from metaflye (n=397) and hifiasm-meta (n=524) vs HiCanu (n=32). None of the bins output by HiCanu are high quality. 

Do you have any advice on how I could improve the performance of HiCanu with my dataset? I based the command I used on the recommendations given in https://canu.readthedocs.io/en/latest/faq.html. 

Command_1:
`canu -p 6730_HET_VAC_adjusted_5m -d 6730_HET_VAC_adjusted_5m genomeSize=5m  maxInputCoverage=10000 corOutCoverage=10000 corMhapSensitivity=high corMinCoverage=0 redMemory=32 oeaMemory=32 batMemory=200 batThreads=16  useGrid=false   -pacbio-hifi ../m64175e_220113_214442.hifi_reads.fastq.gz`

I’ve tried changing the genomeSize option to 100m as I’ve seen that used in a few places, but that made the assembly worse. 

I also tried using the options below, this increased the assembly size a little (68mb) and gave more bins (n=62): Command_2:
`canu -p 6730_HET_VAC_amanda_5m -d 6730_HET_VAC_amanda_5m corMinCoverage=0 corOutCoverage=all corMhapSensitivity=high correctedErrorRate=0.105 genomeSize=5m corMaxEvidenceCoverageLocal=10 corMaxEvidenceCoverageGlobal=10 batMemory=200 useGrid=false   -pacbio-hifi ../m64175e_220113_214442.hifi_reads.fastq.gz`

Thanks for your help,
Laura
[Command_2_log.txt](https://github.com/marbl/canu/files/8679546/Command_2_log.txt)
[Command_2_report.txt](https://github.com/marbl/canu/files/8679547/Command_2_report.txt)
[Command_1_log.txt](https://github.com/marbl/canu/files/8679548/Command_1_log.txt)
[Command_1_report.txt](https://github.com/marbl/canu/files/8679549/Command_1_report.txt)

**Comment by skoren** [comment]
We never spent much time optimizing metagenomic assembly outside of mock communities. Sorry the parameters are so messy since the cover HiFi/CLR/ONT. To simplify your command, you can ignore any parameter related to correction (`corM*`, `corOut*`) since HiFi data doesn't need correction so you can leave off `corMinCoverage=0 corOutCoverage=all corMhapSensitivity=high corMaxEvidenceCoverageLocal=10 corMaxEvidenceCoverageGlobal=10`.

I think the issue is most of your bins here are really more than one strain and so the default parameters are trying to separate them, leading to very low coverage and no assembly. The second command is much better in this respect, it increases the error tolerance over the default for HiFi which will merge some of these variations into a single contig. Flye definitely has a higher error tolerance than HiCanu's default and I expect hifiasm in metagenomic mode does too. Then reason that assembly is still so small is that, without `maxInputCoverage=10000`, canu cut the coverage to only 200x of a 5mb genome so >90% of the data got thrown out. I think if you run the second command with maxInputCoverage it should give reasonable results. 

TLDR, run:`canu -p 6730_HET_VAC_amanda_5m -d 6730_HET_VAC_amanda_5m_cov maxInputCoverage=100000 correctedErrorRate=0.105 genomeSize=5m  batMemory=200 useGrid=false -pacbio-hifi ../m64175e_220113_214442.hifi_reads.fastq.gz` You could also try adding `-untrimmed` to turn on the trimming step and see if it further improves the assembly.

**Comment by Somebodyatthdoor** [comment]
Brilliant, thanks for the advice. I'll try that and let you know what my results are in a couple of weeks.

**Comment by Somebodyatthdoor** [comment]
Hi,
Thanks for your suggestions, I'm now getting a far better assembly that is comparable to what I get with metaflye. Adding in the trimming step seems to make things a little worse, so I'll probably leave that out:

**Without trimming:**
Number of seqs: 10876
N50: 70146
max_length: 2024544

**With trimming:**
Number of seqs: 8089
N50: 67468
max_length: 1795324

Thanks for all your help,
Laura

**Comment by skoren** [comment]
OK, I'll close this issue then.

**Comment by Somebodyatthdoor** [comment]
Hi,

I have discovered one problem that is only occurring with one of my four samples. I seem to be  getting a partioning error.  This doesn't happen when I run with the -trimmed option. 

`canu -p 6737_HOM_CTRL -d 6737_HOM_CTRL maxInputCoverage=100000 correctedErrorRate=0.105 genomeSize=5m batMemory=200 useGrid=false -pacbio-hifi ../m64175e_220115_040821.hifi_reads.fastq.gz`

I get the error:
    cd unitigging
    /exports/cmvm/eddie/eb/groups/watson_grp/software/laura_anaconda/canu_env/bin/utgcns \
      -S ../6737_HOM_CTRL.seqStore \
      -T  ./6737_HOM_CTRL.ctgStore 1 \
      -partition 0.8 1.5 0.1 \
    > ./6737_HOM_CTRL.ctgStore/partitioning.log 2>&1
sh: line 1: ./6737_HOM_CTRL.ctgStore/partitioning.log: No such file or directory

[HiCanu_6737.txt](https://github.com/marbl/canu/files/9071928/HiCanu_6737.txt)

It isn't a big issue, I can  use the trimmed version of the command if necessary, but I have a slight preference for the untrimmed version as it gives me slightly better assemblies for all the other samples. 

Thanks,
Laura

**Comment by skoren** [comment]
I'd guess you ended up with either no contigs or something else went wrong in the bogart step. Can you post the log from there (`unitigger*out`)?

**Comment by Somebodyatthdoor** [comment]
Thanks for pointing me in the right direction. I've had a look in the 4-unitigger directory and the error file has the line 
"-- Not enough memory to load the minimum number of overlaps; increase -M."
So I guess that's solved my question. 

Thanks again,
Laura
